src/raft_log.rs

```diff
@@ -194,6 +194,41 @@ impl<T: Storage> RaftLog<T> {
         0
     }
 
+    /// find_conflict_by_term takes an (`index`, `term`) pair (indicating a conflicting log
+    /// entry on a leader/follower during an append) and finds the largest index in
+    /// log with log.term <= `term` and log.index <= `index`. If no such index exists
+    /// in the log, the log's first index is returned.
+    ///
+    /// The index provided MUST be equal to or less than self.last_index(). Invalid
+    /// inputs log a warning and the input index is returned.
+    ///
+    /// Return (index, term)
+    pub fn find_conflict_by_term(&self, index: u64, term: u64) -> (u64, Option<u64>) {
+        let mut conflict_index = index;
+
+        if index > self.last_index() {
+            warn!(
+                self.unstable.logger,
+                "index({}) is out of range [0, last_index({})] in find_conflict_by_term",
+                index,
+                self.last_index()
+            );
+        }
+
+        loop {
+            match self.term(conflict_index) {
+                Ok(t) => {
+                    if t > term {
+                        conflict_index -= 1
```

Comment:
Is it possible to underflow?

Reply:
if index is out of range, we get `0`, as long as the given `term` is non-negative,  we will return immediately, so no underflow.